tests/test_topline_historical_backfill.py

```diff
@@ -0,0 +1,144 @@
+import json
+import os
+
+from click.testing import CliRunner
+
+import pytest
+from mozetl.topline import historical_backfill as backfill
+from pyspark.sql import SparkSession
+
+
+@pytest.fixture(scope="session")
+def spark(request):
+    spark = (SparkSession
+             .builder
+             .appName("test_topline_historical_backfill")
+             .getOrCreate())
+
+    # teardown
+    request.addfinalizer(lambda: spark.stop())
+
+    return spark
+
+
+default_sample = {
+    "geo": "US",
+    "channel": "nightly",
+    "os": "Windows",
+    "date": "2016-01-01",
+    "actives": "1",
+    "hours": "1",
+    "inactives": "1",
+    "new_records": "1",
+    "five_of_seven": "1",
+    "total_records": "1",
+    "crashes": "1",
+    "default": "1",
+    "google": "1",
+    "bing": "1",
+    "yahoo": "1",
+    "other": "1"
+}
+
+
+def generate_samples(snippets, base_sample):
+    if snippets is None:
+        return [json.dumps(base_sample)]
+
+    samples = []
+    for snippet in snippets:
+        sample = base_sample.copy()
+        sample.update(snippet)
+        samples.append(json.dumps(sample))
+    return samples
+
+
+def snippets_to_df(spark, snippets, base_sample, schema):
+    samples = generate_samples(snippets, base_sample)
+    jsonRDD = spark.sparkContext.parallelize(samples)
+    return spark.read.json(jsonRDD, schema=schema)
+
+
+# does not include all rows from original data
```

Comment:
This comment made sense only once I understood what was going on :). Do you mind quoting "all" to differentiate between 'all rows' and rows containing 'all'